unity-client/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarAudioHandlerLocal.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class AvatarAudioHandlerLocal : MonoBehaviour
+{
+    const float WALK_INTERVAL_SEC = 0.4f;
+    const float RUN_INTERVAL_SEC = 0.27f;
+
+    float intervalTimer = 0f;
+
+    AudioEvent footstepJump;
+    AudioEvent footstepLand;
+    AudioEvent footstepWalk;
+    AudioEvent footstepRun;
+    AudioEvent clothesRustleShort;
+
+    CameraController cameraController;
+
+    private void Start()
+    {
+        DCLCharacterController dclCharacterController = transform.parent.GetComponent<DCLCharacterController>();
+        if (dclCharacterController != null)
+        {
+            dclCharacterController.OnJump += OnJump;
+            dclCharacterController.OnHitGround += OnLand;
+            dclCharacterController.OnMoved += OnWalk;
+        }
+
+        AudioContainer ac = GetComponent<AudioContainer>();
+        footstepJump = ac.GetEvent("FootstepJump");
+        footstepLand = ac.GetEvent("FootstepLand");
+        footstepWalk = ac.GetEvent("FootstepWalk");
+        footstepRun = ac.GetEvent("FootstepRun");
+        clothesRustleShort = ac.GetEvent("ClothesRustleShort");
+    }
+
+    void OnJump()
+    {
+        if (footstepJump != null)
+            footstepJump.Play(true);
+    }
+
+    void OnLand()
+    {
+        if (footstepLand != null)
+            footstepLand.Play(true);
+    }
+
+    // Faking footsteps when in first-person mode, since animations won't play
+    void OnWalk(float distance)
+    {
+        if (cameraController == null)
+        {
+            GameObject camObj = GameObject.Find("CameraController");
+            if (camObj != null)
+                cameraController =  camObj.GetComponent<CameraController>();
+        } else
+        {
+            if (cameraController.currentCameraState.name != "FirstPerson")
+                return;
+
+            if (intervalTimer < 0f)
+            {
+                if (distance > 7.3f)
+                {
+                    if (footstepRun != null)
+                        footstepRun.Play(true);
+                    if (clothesRustleShort != null)
+                        clothesRustleShort.Play(true);
+                    intervalTimer = RUN_INTERVAL_SEC;
+                }
+                else
+                {
+                    if (footstepWalk != null)
+                        footstepWalk.Play(true);
+                    if (clothesRustleShort != null)
+                        clothesRustleShort.PlayScheduled(Random.Range(0.05f, 0.1f));
+                    intervalTimer = WALK_INTERVAL_SEC;
```

Comment:
All these magic numbers: `7.3` `0.05` `0.1` could go to a configuration Scriptable Object so tweaking them wouldnt require code changes.

In any case we can evaluate if it worths the effort and implement it in another PR.

Reply:
I could replace the PlayScheduled method with a few variables in the AudioEvent scriptable object; delayMin and delayMax, and have the regular "Play" event handle the delay. I'll make a note of it and do this on another branch probably.

As for distance > 7.3, I've moved it to the top of the script file as a const for now. Maybe there is some a variable somewhere that I could use instead. The number is used to determine whether the character is running or walking.